### DISCUSSION TITLE
Livewire re-render component: Call to a member function getFilename() on array

### DISCUSSION THREAD

**Discussion by im-denny** [livewire/livewire#2243] [1. Help] [answered]
### Description
I have file upload (using [Livewire fileuploads][1]) functionality with sorting (using [Livewire/sortable][2]), below is more explanation over it.

In the livewire component blade I have for loop over a variable named `$fileuploads`, this variable has two types of data in it, as its edit page, so there will be already existing images and user can upload new ones also. So `$fileuploads` variable have two types of data in it one being `Livewire\TemporaryUploadedFile` another being simple `array` (fetched with laraval eloquent query). In the image tag's `src` attribute I have ternary based on which it'll either display image from `asset('img/filename...')` or Livewire `temporaryUrl()`

Uploading new images and sorting is working fine. When user sorts the images, I have `wire:sortable="changeOrder"`, and this function will simply swap the index of the values in `$fileupload` variable.But there is one edge case in which if user tries to put `Livewire\TemporaryUploadedFile` image on the first place by drag and drop(sorting) then it throws error stating [`Call to a member function getFilename() on array`][5] and component doesn't re-renders, in each and every other cases it works fine and component re-renders successfully. 

[Here is sneak peek into UI][6]

  [1]: https://laravel-livewire.com/docs/2.x/file-uploads#preview-urls
  [2]: https://github.com/livewire/sortable
  [5]: https://i.stack.imgur.com/bJRkN.png
  [6]: https://drive.google.com/file/d/10sVkwguH_b0xWuzdpPNvwqeaCof_8rJp/view?usp=sharing

**Context**
- Livewire version: [2.3.5]
- Laravel version: [7.30.0]
- Alpine version: [2.7.0]
- Browser: [Chrome]

**Comment by joshhanley** [comment]
@im-denny can you share your component class and blade view so we can have a look?

**Comment by im-denny** [comment]
Hi @joshhanley  

Yes, I have updated my [description](https://github.com/livewire/livewire/discussions/2243#discussion-78220)

**Comment by some-daan** [comment]
I run into the same problem. Any update on this issue?

**Context**
Livewire version [2.4.4]
Laravel version [8.29.0]
Alpine version [2.7.3]
Browser [Chrome]

**Comment by some-daan** [comment, marked as answer]
@im-denny This is fixed in v2.8.0
